### ISSUE TITLE
Wrong size when creating a box

### ISSUE THREAD

**Issue by xTop-SecreTx** [qbart/isometric-pixel-art#3]
Hi i would love to use your tool. I was looking for something exactly like this.

The problem is when i create for example a box of 32x32x32 pixel the image should be 64x64 pixel, but it isn't. Actually it is 62x62.

I need this tile to fit in my game so i can't use your tool :(
![image](https://user-images.githubusercontent.com/12834038/37616350-dc5f659e-2baf-11e8-91fe-748663c42ff8.png)

**Comment by qbart** [comment]
Hi there,

Yes, that's true, my tool won't generate sizes you want 😢  because it was designed to draw tiles differently but I will try to explain "why?".

When you do isometric you need consider how you will connect tiles into the final map and there are different methods for that. Please take a look (I simplified sizes and limited myself to 2D only but general rules apply for 3D as well)

![untitled](https://user-images.githubusercontent.com/235847/37621589-a27f5b12-2bbf-11e8-8f32-54d6a214aa4a.png)

What you are doing with your game is the solution `A`. The reason why I didn't follow it is because when you look at the map and tile edge connections you will see (you need to zoom it) that when the lines cross each other they create extra bold pixels - personally I find this ugly. What is great though that when rendering the map calculations stay simple (power of 2).

Next solution `B` is the one I am using, it's my favourite and generates perfect edges, just look at them how they seamlessly connect but math behind it is not as beautiful as it would be in other solutions.

Final solution `C` I think it would be the most popular to use in games (although I can be wrong).
The reason for it is the tile is always increased by 1px for bottom edges and when the map is rendered calculations are as simple as solution `A` but final map looks much nicer.

To sum up, the best solution for me was `B` and that's why it was implemented.
If you don't care about the edges maybe solution `A` is enough.
If you decide to choose `B` and use the tool I think it should be easy to change the code that generates the map in you game but I should also mention that I no longer maintain this project ⚠️, simply don't have time for it - so choose wisely.

Good luck with your game 🥇

**Comment by xTop-SecreTx** [comment]
Thank you very much for your extensive explanation. I didn't know that
there are these 3 ways of building isometric tiles.
I'm quite a noob in programming such maps and there is not too much on
google about isometric sprite maps. I find it hard to create sprites for it
so i found your tool. What a pity i can't use it actually. I'll see if i
can stay on method A. Wish there would be a program that can create
isometric basic forms so i only have to "paint" the content. I use GIMP and
drawing these 2 pixel lines is not that easy as i would like it.
Thanks for clarification.


Regards,
Daniela


2018-03-19 22:29 GMT+01:00 Bartłomiej Wójtowicz <notifications@github.com>:
